File: bindings/csharp/FastParse/LanguageExtensionLoadResult.cs
namespace FastParse;

/// <summary>
/// Result returned after loading a native FastParse language extension.
/// </summary>
public sealed class LanguageExtensionLoadResult
{
    /// <summary>Canonical parse language registered by the extension.</summary>
    public string Language { get; init; } = string.Empty;

    /// <summary>Human-readable language name from the extension descriptor.</summary>
    public string DisplayName { get; init; } = string.Empty;
}

File: bindings/java/fastparse/src/main/java/dev/fastparse/LanguageExtensionLoadResult.java
package dev.fastparse;

/** Result returned after loading a FastParse language extension. */
public final class LanguageExtensionLoadResult {
    private final String language;
    private final String displayName;

    public LanguageExtensionLoadResult(String language, String displayName) {
        this.language = language == null ? "" : language;
        this.displayName = displayName == null ? "" : displayName;
    }

    public String getLanguage() { return language; }
    public String getDisplayName() { return displayName; }
}
